src/models/event.js

```diff
@@ -1098,6 +1098,7 @@ utils.extend(MatrixEvent.prototype, {
             origin_server_ts: this.getTs(),
             unsigned: this.getUnsigned(),
             room_id: this.getRoomId(),
+            txn_id: this.getTxnId(),
```

Comment:
This JSON view (which drives view source features) is meant to "simulate" the format of events flowing through the protocol, so I am a bit unsure about adding this here... It might be useful, but I don't think it matches the format... 🤔

Reply:
If the goal of adding this here is purely debugging, then it might be better to extend the view source display in React SDK to show it somewhere outside of the JSON directly.

Reply:
it isn't for debugging though, this JSON representation is what is serialised for persistence.

Reply:
Well, it wasn't intended for persistence when I first added it... 😬

Reply:
Well it being used for such as of https://github.com/matrix-org/matrix-js-sdk/pull/1655/files#diff-0b31693bb53246dda6511feb2fecc466483c08239e007ce35e067eefb13e967cR403

Reply:
Maybe we need separate functions for persisting an `Event` model object vs. showing a view source formatted view? Previously we were cheating, because the `toJSON` was never used for persistence like this until https://github.com/matrix-org/matrix-js-sdk/pull/1655.

Reply:
The `toJSON` function also currently "makes up" whole extra blocks with `decrypted` and `encrypted` fields for all encrypted events... I don't think that's what we'd want to parse by into the event model when restoring it.

Reply:
nor would we really want to persist in localStorage decrypted events

Reply:
Thank you for the comments here! I'll look into that part a bit more thoroughly. 
Totally missed the "only for debugging" comment. I'll see what is usually done in the rest of the app and try to mimic that